src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover.js

```diff
@@ -600,47 +657,23 @@ function discoverController(
           },
         })
       );
-
-      $scope.$watchCollection('state.sort', function(sort) {
-        if (!sort) return;
-
-        // get the current sort from searchSource as array of arrays
-        const currentSort = getSortArray($scope.searchSource.getField('sort'), $scope.indexPattern);
-
-        // if the searchSource doesn't know, tell it so
-        if (!angular.equals(sort, currentSort)) $fetchObservable.next();
-      });
-
-      // update data source when filters update
-
       subscriptions.add(
-        subscribeWithScope($scope, filterManager.getUpdates$(), {
+        subscribeWithScope($scope, timefilter.getRefreshIntervalUpdate$(), {
           next: () => {
-            $scope.updateDataSource().then(function() {
-              $state.save();
-            });
+            $scope.refreshInterval = timefilter.getRefreshInterval();
```

Comment:
Why do you keep the refresh interval on the scope, instead of always fetching it when you want to sync state ?

Reply:
true, thx, removed!